File: src/main/java/ru/grabovsky/recordkeeping/repositories/UserRepository.java
package ru.grabovsky.recordkeeping.repositories;

import org.springframework.data.jpa.repository.JpaRepository;
import ru.grabovsky.recordkeeping.models.entities.User;

import java.util.Optional;

public interface UserRepository extends JpaRepository<User, Long> {
    Optional<User> findByUsername(String username);
    boolean existsUserByUsernameIgnoreCase(String username);
    boolean existsUserByEmailIgnoreCase(String email);
    Optional<User> findByEmail(String email);
}

File: core/src/main/kotlin/ru/grabovsky/recordkeeping/core/repositories/db/UserRepository.kt
package ru.grabovsky.recordkeeping.core.repositories.db

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import ru.grabovsky.recordkeeping.core.entity.db.User
import java.util.*

/**
 *
 *
 * @author GrabovskyAlexey
 * @date 03.01.2024
 */
@Repository
interface UserRepository : JpaRepository<User, Long> {
    fun findByUsername(username: String) : Optional<User>
    fun findByEmail(email: String): Optional<User>
    fun existsByEmailIgnoreCase(email: String): Boolean
    fun existsByUsernameIgnoreCase(username: String): Boolean
}